### DISCUSSION TITLE
question about sqrtPriceLimitX96

### DISCUSSION THREAD

**Discussion by chrisyy2003** [Jeiwan/uniswapv3-book#63] [Milestone 0. Introduction]
I have a question about `sqrtPriceLimitX96`, that is, when `zeroforone` is true, the price moves to the right, and then through `sqrtPriceLimitX96` to control whether to reach the upper limit, but why is `sqrtPriceLimitX96` required to be less than `slot0Start.sqrtPriceX96` at the beginning of swap, in my opinion, `sqrtPriceLimitX96` should be greater than `slot0Start.sqrtPriceX96` is right, who can help me solve the problem, thanks

**Comment by OoXooOx** [comment]
zeroForOne == true => price will go down => sqrtPriceLimitX96 must be <then slot0Start.sqrtPriceX96. All logical. What is question?
